lib/backend/k8s/k8s_test.go

```diff
@@ -1199,6 +1199,120 @@ var _ = Describe("Test Syncer API for Kubernetes backend", func() {
 		})
 	})
 
+	It("should handle patching a Pod with our IP address annotation", func() {
+		pod := &k8sapi.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-syncer-basic-pod",
+				Namespace: "default",
```

Comment:
Probably worth also having a test with pre-existing annotations, or racing requests to update annotations, to make sure we've got our patch logic right (e.g. we don't overwrite other annotations).

Reply:
Added a test with another annotation.  I experimented a bit with races yesterday.  I found that even an `UpdateStatus()` can clobber the annotation we add if you don't include the revision.  If you include a revision then the request is rejected because the patch makes your copy of the data stale.  I worked that behaviour into the test.

Upshot is, if we race with someone using the API correctly then our patch makes their request fail.  If they're using it incorrectly and not supplying a revision then they do clobber us.

Guess we could do a test of multiple patches being applied at the same time but that really would just be testing the API server applies patches correctly.